Add Get Restaurant/s by Name Endpoint  #13

Khilan10 opened the pull request:
- Add mapping /restaurant/name/{reastaurant_name} to get Restaurants by name

- Add named query in Restaurant Entity

Khilan10 committed:
Add Get Restaurant/s by Name Endpoint  #13

- Add mapping /restaurant/name/{reastaurant_name} to get Restaurants by name

- Add named query in Restaurant Entity

pavankarthiktanikanti reviewed (changes requested):
- Resolve conflicts in RestExceptionHandler

- Change the name of resource in request mapping to {restaurant_name}

- Test the endpoint error case when restaurant name is empty 

- Use String Builder for concatenation of String

- Test the endpoint by sending the request in all lowercase

pavankarthiktanikanti committed:
Resolve conflicts with master branch #13

pavankarthiktanikanti committed:
Merge branch 'master' into feature/restaurant

Khilan10 committed:
Update request mapping name #13

- Change the name of resource in request mapping to {restaurant_name}

- Use String Builder for concatenation of String

Khilan10 committed:
Add Endpoint to get Restaurants based upon the category uuid #14

- Add mapping /restaurant/category/{category_id} to get list of restaurants based upon category uuid

- Add named Query in RestaurantCategoryEntity

Khilan10 committed:
Add Endpoint to get Restaurant based upon the restaurant uuid

- Add mapping /api/restaurant/{restaurant_id} to get restaurants based upon restaurant uuid

- Add populateRestaurantDetailResponse method in Restaurant Controller to get Restaurant Detail Response from Restaurant Entity

- Add named query in Restaurant Category Entity

Khilan10 committed:
Add Endpoint to update the customer rating for a particular a restaurant #16

- Add mapping /api/restaurant/{restaurant_id} to update customer rating for a restaurant

- Add invalidRatingException in Rest Exception Handler

Khilan10 committed:
Updated Parameter in set Id for restaurantUpdateResponse so that test case pass #16

Khilan10 committed:
Updating path variable name #13

pavankarthiktanikanti reviewed (changes requested):
- line 104 Restaurant Controller
	- rename restaurantsListByUuid as per the endpoint

- line 46,67,87 RestaurantService.java use util method isInvalid

- line 108 RestaurantService.java use util method !isInvalid

pavankarthiktanikanti committed:
Merge branch 'master' into feature/restaurant

Khilan10 committed:
Rename restaurantsListByUuid and use util method isInvalid #13 #14 #16 #45

- rename restaurantsListByUuid to  restaurantsListByCategory as per the endpoint in Restaurant controller

- use util method isInvalid in Restaurant service

- add getItemsByPopularity in Item Dao that was deleted by mistake
